joblib/numpy_pickle.py

```diff
@@ -389,29 +385,38 @@ def dump(value, filename, compress=0, cache_size=100, protocol=None):
     using compression can significantly slow down loading. In
     addition, compressed files take extra extra memory during
     dump and load.
+
     """
     if compress is True:
-        # By default, if compress is enabled, we want to be using 3 by
-        # default
+        # By default, if compress is enabled, we want to be using 3 by default
         compress = 3
+
     if not isinstance(filename, _basestring):
         # People keep inverting arguments, and the resulting error is
         # incomprehensible
         raise ValueError(
-              'Second argument should be a filename, %s (type %s) was given'
-              % (filename, type(filename))
-            )
+            'Second argument should be a filename, %s (type %s) was given'
+            % (filename, type(filename))
+        )
+
+    if cache_size is not None:
+        # Cache size is deprecated starting from version 0.10
+        warnings.warn("Cache size is deprecated and will be ignored.",
```

Comment:
Use " 'cache_size' " to match the parameter name. Bonus points if you add the value it was set to in the warning. Maybe something like:

```python
"Please do not set 'cache_size' in joblib.dump, this parameter has no effect and will be removed. "
"You used 'cache_size={0}'".format(cache_size)
```